doc/source/timeseries.rst

```diff
@@ -557,6 +557,50 @@ We are stopping on the included end-point as it is part of the index
    dft2 = dft2.swaplevel(0, 1).sort_index()
    dft2.loc[idx[:, '2013-01-05'], :]
 
+.. versionadded:: 0.21.1
+
+``DatetimeIndex`` partial string indexing can be used with naive datetime-like labels when the ``DatetimeIndex`` has no timezone set.
+If a timezone is provided by the label, the datetime index is assumed to be UTC and a ``UserWarning`` is emitted.
+
+.. note::
+
+   This both works with ``pd.Timestamp`` and strings
+
+.. ipython:: python
+   :okwarning:
+
+   first_january_implicit_utc = pd.date_range('2016-01-01T00:00', '2016-01-01T23:59',
+                                              freq='T')
+
+   df = pd.DataFrame(index=first_january_implicit_utc,
+                     data=np.arange(len(first_january_implicit_utc)))
+
+   df
+
+   four_minute_slice = df["2016-01-01T00:00-02:00":"2016-01-01T02:03"]
```

Comment:
Yes you are right. What is the consequence in your eyes? I just want the timezones to work, that is my only desire.

Reply:
There is no consequence for the behaviour, so this PR will your usecase, But for the example in the docs, we should make a clear one. So either I would make this actual partial slicing, or move this section to somewhere else

Reply:
Then better move it because the timezones can not always be parsed, e.g. for months still UTC will be assumed as it goes through another path.

Reply:
No, you can just edit the example a little bit. For example keep the minute resolution, and use strings with only hours (instead of the minutes now, and that still provides ability to specify time zone), or change the resolution of the df to seconds, and keep the strings as they are. Note you can do eg each 30s to avoid that selecting some minutes results in many rows.